### DISCUSSION TITLE
How to filter in array of object

### DISCUSSION THREAD

**Discussion by sabringhouili** [pocketbase/pocketbase#3346] [Q&A]
Hello,

Is it possible to filter records according to the value of an array of objects in a json field. For example, if a table has a position field with the following values :
[
  {
    "isoCode": "fr",
    "value": "Product Owner"
  },
  {
    "isoCode": "en",
    "value": "Product Owner"
  }
]
Is it possible to obtain records equal to position.value="Product Owner".

Sincerely,

**Comment by ganigeorgiev** [comment]
No, this is not supported at the moment.

For `json` fields we currently support only direct [`json_extract`-like](https://www.sqlite.org/json1.html#jex) retrieval using dot-notation, meaning that you need to know the exact path of the value you want to access, eg. `position.1.value = "Product Owner"`.

Depending on your use case, if you can't modify your data structure and use a separate collection, an alternative could be to use a loose like/contains comparison using the `~` operator, something like:

```js
position ~ '"Product owner"'
``` 
_(the above will produce a `LIKE` constraint for the raw serialized json value, aka. `'[{"isoCode": "fr", "value": "Product Owner"}, ...]' LIKE '%"Product Owner"%'`)_
